modules/test-runner/src/swap/swap.test.ts

```diff
@@ -130,4 +130,123 @@ describe("Swaps", () => {
     );
     expect(postSwapFreeBalanceTokenNode).toBeBigNumberEq(swapAmount);
   });
+
+  test("Bot A tries to swap with invalid from token address", async () => {
+    // client deposit and request node collateral
+    await clientA.deposit({ amount: parseEther("0.01").toString(), assetId: AddressZero });
+    await clientA.requestCollateral(tokenAddress);
+
+    const swapRate = await clientA.getLatestSwapRate(AddressZero, tokenAddress);
```

Comment:
we should be checking this swap rate is as expected

Reply:
How exactly can we check swap rate is as expected? This fn pulls it from node lol